src/mlwm/build_inference_artifact.py

```diff
@@ -0,0 +1,350 @@
+"""
+Contents required when package to an inference artifact:
+
+- Model checkpoint, provided by a path to the CLI. Written to `checkpoint.pkl`
+- Extraction of statistics from training dataset. Paths to datasets inferred
+  from datastore configs link reference in neural-lam config provided to cli.
+  Written to `stats/{datastore_name}.stats.zarr`
+- Command line arguments that training was run with. Expected to be in
+  `training_cli_args.yaml`
+"""
+import os
+import shutil
+import sys
+import tempfile
+import zipfile
+from pathlib import Path
+
+import dotenv
+import mllam_data_prep as mdp
+import xarray as xr
+import yaml
+from loguru import logger
+from tqdm import tqdm
+from upath import UPath
+
+from . import __version__
+
+DEFAULT_ARTIFACT_PATH_FORMAT = (
+    "s3://mlwm-artifacts/inference-artifacts/{artifact_name}.zip"
+)
+
+dotenv.load_dotenv()
+
+ARTIFACT_PATH_FORMAT = os.environ.get(
+    "ARTIFACT_PATH_FORMAT", DEFAULT_ARTIFACT_PATH_FORMAT
+)
+
+TRAINING_CLI_ARGS_FILE = "training_cli_args.yaml"
+
+
+def _find_datastore_paths(nl_config_path: str):
+    with open(nl_config_path, "r") as f:
+        nl_config = yaml.safe_load(f)
+
+    def _make_abspath(datastore_path: str):
+        """
+        Make the path absolute if it is not already. If the path is relative,
+        it is made absolute with respect to the neural-lam config file.
+        """
+        # if the path is absolute, return it as is
+        if Path(datastore_path).is_absolute():
+            return datastore_path
+        return Path(nl_config_path).parent / datastore_path
+
+    datastore_paths = {}
+    if "datastore" in nl_config:
+        datastore_paths[None] = _make_abspath(
+            nl_config["datastore"]["config_path"]
+        )
+    elif "datastores" in nl_config:
+        for name, datastore_config in nl_config["datastores"].items():
+            datastore_paths[name] = _make_abspath(
+                datastore_config["config_path"]
+            )
```

Comment:
Can we have multiple datastores?

Reply:
Yes, in the code on the `neural-lam` research branch the neural-lam config contains a datastore for the domain interior and boundary separately. So I was trying to to anticipate that here 😆

Reply:
I will add a docstring and explain this!

Reply:
done in https://github.com/dmidk/mlwm-deployment/pull/1/commits/ca3cd43a1a212ed79ad5b4c86f82fa37230e80ae